File: Lab_07.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *file;
    char ch;

    // --------- Simulate ClassNotFound by trying to open a non-existent file ---------
    file = fopen("non_existing_class.txt", "r");
    if (file == NULL) {
        printf("ClassNotFound simulation: File 'non_existing_class.txt' not found.\n");
    } else {
        fclose(file);
    }

    // --------- EOFException Simulation ---------
    file = fopen("input_Lab07.txt", "r");

    if (file == NULL) {
        printf("EOFException simulation: 'input.txt' file not found. Create this file to test EOF.\n");
    } else {
        printf("Reading characters until EOF:\n");

        while ((ch = fgetc(file)) != EOF) {
            putchar(ch);
        }

        printf("\nReached end of file (EOF).\n");
        fclose(file);
    }

    return 0;
}

File: Lab_07.java
import java.io.*;

public class Lab_07 {
    public static void main(String[] args) {
        // ClassNotFound simulation
        File file = new File("non_existing_class.txt");
        if (!file.exists()) {
            System.out.println("ClassNotFound simulation: File 'non_existing_class.txt' not found.");
        }

        // EOF simulation using BufferedReader
        try (BufferedReader br = new BufferedReader(new FileReader("input_Lab07.txt"))) {
            System.out.println("Reading characters until EOF:");
            int ch;
            while ((ch = br.read()) != -1) {
                System.out.print((char) ch);
            }
            System.out.println("\nReached end of file (EOF).");
        } catch (IOException e) {
            System.out.println("IOException occurred: " + e.getMessage());
        }
    }
}
